maraboupy/MarabouNetworkONNX.py

```diff
@@ -443,22 +444,30 @@ def batchNorm(self, node):
         scales = self.constantMap[node.input[1]].reshape(-1)
         biases = self.constantMap[node.input[2]].reshape(-1)
         input_means = self.constantMap[node.input[3]].reshape(-1)
-        input_vars = self.constantMap[node.input[4]].reshape(-1)
+        input_variances = self.constantMap[node.input[4]].reshape(-1)
+
+        if not makeEquations:
+            return
+
+        numChannels = len(scales)
 
         # Get variables
-        inputVars = self.varMap[inputName].reshape(-1)
-        outputVars = self.makeNewVariables(nodeName).reshape(-1)
-        assert len(inputVars) == len(outputVars)
+        inputVars = self.varMap[inputName].reshape(numChannels, -1)
+        outputVars = self.makeNewVariables(nodeName).reshape(numChannels, -1)
+        assert(inputVars,shape == outputVars.shape)
 
-        for i in range(len(inputVars)):
-            # Add equation
-            # To know this computation, 
-            # refer to https://github.com/onnx/onnx/blob/master/docs/Operators.md#batchnormalization.
-            e = MarabouUtils.Equation()
-            e.addAddend(-1, outputVars[i])
-            e.addAddend(1 / np.sqrt(input_vars[i] + epsilon) * scales[i], inputVars[i])
-            e.setScalar(input_means[i] / np.sqrt(input_vars[i] + epsilon) * scales[i] - biases[i])
-            self.addEquation(e)      
+        numInputs = inputVars.shape[1]
+
+        for i in range(numChannels):
+            for j in range(numInputs):
+                # Add equation
+                # To know this computation,
+                # refer to https://github.com/onnx/onnx/blob/master/docs/Operators.md#batchnormalization.
+                e = MarabouUtils.Equation()
+                e.addAddend(-1, outputVars[i][j])
+                e.addAddend(1 / np.sqrt(input_variances[i] + epsilon) * scales[i], inputVars[i][j])
+                e.setScalar(input_means[i] / np.sqrt(input_variances[i] + epsilon) * scales[i] - biases[i])
+                self.addEquation(e)
```

Comment:
It looks like you are trying to expand it to multi channels rather than multi dimensions.

Reply:
Sorry, never mind this.
Input is flattened by `self.varMap[inputName].reshape(numChannels, -1)`.